autotest/gdrivers/cderived.py

```diff
@@ -0,0 +1,93 @@
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+import sys
+from osgeo import gdal
+
+sys.path.append( '../pymod' )
+
+import gdaltest
+
+###############################################################################
+# Test opening a L1C product
+
+
+def cderived_test():
+    filename = "data/cfloat64.tif"
+    gdal.ErrorReset()
+    ds = gdal.Open(filename)
+    if ds is None or gdal.GetLastErrorMsg() != '':
+        gdaltest.post_reason('fail')
+        return 'fail'
+    got_dsds = ds.GetMetadata('DERIVED_SUBDATASETS')
+
+    expected_dsds = {'DERIVED_SUBDATASET_0_NAME' : 'DERIVED_SUBDATASET:AMPLITUDE:data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_0_DESC' : 'Amplitude of input bands from data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_1_NAME' : 'DERIVED_SUBDATASET:PHASE:data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_1_DESC' : 'Phase of input bands from data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_2_NAME' : 'DERIVED_SUBDATASET:REAL:data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_2_DESC' : 'Real part of input bands from data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_3_NAME' : 'DERIVED_SUBDATASET:IMAG:data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_3_DESC' : 'Imaginary part of input bands from data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_4_NAME' : 'DERIVED_SUBDATASET:CONJ:data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_4_DESC' : 'Conjugate of input bands from data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_5_NAME' : 'DERIVED_SUBDATASET:INTENSITY:data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_5_DESC' : 'Intensity (squared amplitude) of input bands from data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_6_NAME' : 'DERIVED_SUBDATASET:LOGAMPLITUDE:data/cfloat64.tif',
+                     'DERIVED_SUBDATASET_6_DESC' : 'log10 of amplitude of input bands from data/cfloat64.tif'}
+
+    if got_dsds != expected_dsds:
+        gdaltest.post_reason('fail')
+        import pprint
+        pprint.pprint(got_dsds)
+        return 'fail'
+    
+
+    for (key,val) in expected_dsds.iteritems():
+        if key.endswith('_NAME'):
+            ds = gdal.Open(val)
+            if ds is None or gdal.GetLastErrorMsg() != '':
+                gdaltest.post_reason('fail')
+                return 'fail'
+    
```

Comment:
Might be good to test also :
- that the geotransform and projection string are well transformed
- to verify the checksum on one subdataset to check that the bands are correctly instanciated